### PR TITLE
AV-203582 add amkouuid in healthmonitor name for passthrough route

### PR DESCRIPTION
**Pull request by arihantg**
Tested on open shift setup by creating a passthrough route.  Health monitor amko--passthrough-hm-tcp was replaced by hm with name amko--passthrough-hm-tcp-6152369d-f751-11ee-999e-12c08ce663db

### CODE REVIEW

**Comment by arihantg** [comment]
build amko

**Comment by arihantg** [comment]
build amko

**Comment by arihantg** [comment]
build amko

**Comment by arihantg** [comment]
build amko

File: gslb/nodes/avi_model_nodes.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func (hm HealthMonitor) GetHMDescription(gsName string, template *string) []stri
 	descPrefix := CreatedByAMKO + ", gsname: " + gsName
 	desc = append(desc, descPrefix)
 	if hm.Type == NonPathHM {
-		if hm.Name == gslbutils.SystemGslbHealthMonitorPassthrough {
+		if hm.Name == gslbutils.SystemGslbHealthMonitorPassthrough+gslbutils.AMKOControlConfig().GetAMKOUUID() {
```
Comment (akshayhavile) [inline gslb/nodes/avi_model_nodes.go:158]: What will be the impact on AMKOUUID change?

Comment (akshayhavile) [reply gslb/nodes/avi_model_nodes.go:158]: Did we test upgrade scenario?

**Review by akshayhavile — COMMENTED** [review]
@arihantg: It will be good to add retry logic for HM where if we get `already exists error` for AMKO based HM., we will convert `POST` to `PUT`.

**Comment by arihantg** [comment]
> @arihantg: It will be good to add retry logic for HM where if we get `already exists error` for AMKO based HM., we will convert `POST` to `PUT`.

We do it normally also for existing objects without retry. Here additional constraint is that  this HM was created by another AMKO instance so we can't modify it from a different amko instance.

File: gslb/nodes/avi_model_nodes.go (hunk 1 above)
Comment (arihantg) [reply gslb/nodes/avi_model_nodes.go:158]: on amkouuid change new hm will be created with name including the new amkouuid in name. 
Tested the upgrade scenario where gs was created with  amko--passthrough-hm-tcp  as hm attached. on upgrade new hm with new name was attached to GS.

**Review by akshayhavile — APPROVED** [review]

**Pull request merged**